agent/replacecall/src/main/java/randoop/instrument/ReplaceCallAgent.java

```diff
@@ -260,6 +339,98 @@ private static Set<String> loadExclusions(@Owning Reader exclusionReader, String
     return excludedPackagePrefixes;
   }
 
+  /**
+   * Parses an exclusion string into a {@link MethodSignature}.
+   * <p>Accepts the colon/JVM-descriptor form: {@code pkg.Clazz.method:(...)}.
+   * Examples: {@code java.lang.System.exit:(I)V},
+   * {@code java.util.Objects.requireNonNull:(Ljava/lang/Object;)Ljava/lang/Object;}
+   * <p>Returns null if parsing fails or if the input is not in the colon form.
+   *
+   * @param text the string to parse
+   * @return the parsed {@link MethodSignature} or null if parsing fails
+   */
+  private static @Nullable MethodSignature parseExcludedSignature(String text) {
+    String s = text.trim();
+
+    int colon = s.indexOf(':');
+    if (colon > 0 && s.indexOf('(') > colon) {
+      String ownerAndName = s.substring(0, colon);
+      int dot = ownerAndName.lastIndexOf('.');
+      if (dot > 0) {
+        String owner = ownerAndName.substring(0, dot);
+        String name  = ownerAndName.substring(dot + 1);
+        String jvmDesc = s.substring(colon + 1); // "(..)[ret]"
+        String paramList = descriptorParamsToBinaryNames(jvmDesc);
+        if (paramList != null) {
+          try {
+            return MethodSignature.of(owner + "." + name + "(" + paramList + ")");
+          } catch (IllegalArgumentException ignored) {
+            return null;
+          }
+        }
+      }
+    }
+
+    // Any other format is not supported.
+    return null;
+  }
+
+  /**
+   * Converts a JVM method descriptor to a comma-separated list of binary names suitable for
+   * {@link MethodSignature#of(String)}. For example,
+   *   "(I)V" -> "int"
+   *   "([Ljava/lang/String;I)V" -> "java.lang.String[], int"
+   * Returns null if the input doesn't look like a valid method descriptor.
+   */
+  private static @Nullable String descriptorParamsToBinaryNames(String methodDescriptor) {
```

Comment:
The implementation of this method can be shorter by using methods from [reflection-util](https://plumelib.org/reflection-util/api/org/plumelib/reflection/Signatures.html).